### ISSUE TITLE
Compilation error after updating to Erlang OTP 21 - "Hook for compile failed!"

### ISSUE THREAD

**Issue by darrenklein** [davisp/jiffy#174]
Hello, thanks so much for your work on this project.  I've recently upgraded from Erlang 20 -> 21, and I'm having an issue compiling jiffy that I don't quite understand, was hoping you could point me in the right direction.

First off, the error:

```
===> Compiling jiffy
Compiling c_src/decoder.c
sh: line 0: exec: darren@work.com: not found
===> Hook for compile failed!

==> my_app
** (Mix) Could not compile dependency :jiffy, "/Users/darrenklein/.mix/rebar3 bare compile --paths "/Users/darrenklein/Documents/work/my_app/_build/dev/lib/*/ebin"" command failed. You can recompile this dependency with "mix deps.compile jiffy", update it with "mix deps.update jiffy" or clean it with "mix deps.clean jiffy"
```

Where would it be pulling that email address, `darren@work.com` from, and what is it trying to do with it?  I have also tried to deploy this app to Heroku, and receive the same error message there (albeit with different paths in the error, of course).

I am using the most recent commit of jiffy, `ee8b48cb448a999e1fddbb5e2e32e7dfe33acf5c`.

Ok, a little context to clear things up - my local development environment:

My machine is a Macbook Pro, running Mojave 10.14.3.

I'm using jiffy in a Phoenix app.  I have installed Elixir and Erlang via Homebrew, and the output of `elixir --version` is

```
Erlang/OTP 21 [erts-10.2.3] [source] [64-bit] [smp:4:4] [ds:4:4:10] [async-threads:1] [hipe] [dtrace]

Elixir 1.8.1 (compiled with Erlang/OTP 21)
```

as far as Heroku goes, the versions of Erlang and Elixir are the same.

Is there any other info that I can provide to help debug this?  Any clarification you can offer?

Again, many thanks!

**Comment by davisp** [comment]
Hiya!

First things first, can you try cloning jiffy directly and compiling it in its own repo? A simple ‘make’ in the jiffy clone should be good. I’m not the most hip to the intricacies of mix and friends so I’d like to narrow it down to whether its the jiffy itself or some interaction between the build systems.

Jiffy does use a small binary thats checked into the repo called ‘enc’ which is basically rebar2 stripped down to just a NIF compiler. Its possible something in your local build environment is confusing the invocation of that compile hook.

If Jiffy doesn’t compile oitside the tree we can look harder at that. If it does then we can start poking at how mix is invoking dependency builds tonsee if something has changed there recently.

**Comment by davisp** [comment]
I should also add that I know Jiffy compiles fine on Erlang 21 but don’t always keep up with Elixir releases. So my money is on something in the build system interacting weirdly or possibly (though less likely given the Heroku reproduction) your specific build environment.

**Comment by darrenklein** [comment]
Ah, thank you so much for getting right on top of this and helping me - I managed to figure out what was going on, and it was a problem of my own making.

In both my local and deployment environments, I had set an environment variable named `CC`, which was intended to be a Cc'd recipient of emails that my server sends.  However, unbeknownst to me, the `CC` environment variable is used by the Unix shell to define the path to a compiler... so I was messing that up!

I hadn't needed to recompile Jiffy in either environment for quite some time and, during that dry spell, had started setting the `CC` variable - so it is doubtful that this problem had anything to do with the Erlang version bump.  When I did the version update, I had some issues with other dependencies and just ended up clearing them all out and starting fresh, and that's when I dropped this piano on my foot - the update/recompile exposed a flaw that was already there.

Thanks so much for your quick response on this, and all of your hard work!
